pandas/core/frame.py

```diff
@@ -6426,19 +6431,68 @@ def sort_values(  # type: ignore[override]
         else:
             return result.__finalize__(self, method="sort_values")
 
-    @deprecate_nonkeyword_arguments(version=None, allowed_args=["self"])
+    @overload
     def sort_index(
         self,
+        *,
```

Comment:
Should you add the `DataFrameT` annotation to `self` here?

Reply:
No, DataFrameT isn't returned here - would be an unbound TypeVar.